tests/test_boot.py

```diff
@@ -140,23 +138,73 @@ def _cleanup_boom_root_path(self, boot_path):
     def setUp(self):
         log.debug("Preparing %s", self._testMethodName)
         boom.set_boot_path("/boot")
+
+
+    def tearDown(self):
+        log.debug("Tearing down %s", self._testMethodName)
+        boom.set_boot_path(BOOT_ROOT_TEST)
+        self._clear_fstab()
+
+
+@unittest.skipIf(not have_root(), "requires root privileges")
+class BootTestsNonRoot(BootTestsBase):
+
+    def setUp(self):
+        self.volumes = ["root", "home", "var"]
+        self.thin_volumes = ["opt", "srv"]
+        self.boot_volumes = [
+                ("root", "/"),
+                ("home", "/home"),
+                ("var", "/var"),
+        ]
+        self._set_fstab()
         self._lvm = LvmLoopBacked(self.volumes, thin_volumes=self.thin_volumes)
-        snapset_name = "bootset0"
-        snapset_time = 1707923080
         for origin, mp in self.boot_volumes:
+            if origin == "root":
+                continue
             self._lvm.create_snapshot(
                 origin,
                 format_snapshot_name(
-                    origin, snapset_name, snapset_time, encode_mount_point(mp)
+                    origin, self.snapset_name, self.snapset_time, encode_mount_point(mp)
                 ),
             )
         self.manager = snapm.manager.Manager()
+
+
+    def tearDown(self):
+        log.debug("Tearing down %s", self._testMethodName)
+        self._lvm.destroy()
+
+    def test_create_snapshot_boot_entry(self):
+        self.manager.create_snapshot_set_boot_entry(name=self.snapset_name)
+
+        # Clean up boot entry
+        self.manager.delete_snapshot_sets(snapm.Selection(name=self.snapset_name))
+@unittest.skipIf(not have_root(), "requires root privileges")
+class BootTestsWithRoot(BootTestsBase):
+
+    def setUp(self):
+        self.volumes = ["root", "home", "var"]
+        self.thin_volumes = ["opt", "srv"]
+        self.boot_volumes = [
+                ("root", "/"),
+                ("home", "/home"),
+                ("var", "/var"),
+        ]
         self._set_fstab()
+        self._lvm = LvmLoopBacked(self.volumes, thin_volumes=self.thin_volumes)
+        for origin, mp in self.boot_volumes:
+            self._lvm.create_snapshot(
+                origin,
+                format_snapshot_name(
+                    origin, self.snapset_name, self.snapset_time, encode_mount_point(mp)
+                ),
+            )
+        self.manager = snapm.manager.Manager()
+
 
     def tearDown(self):
         log.debug("Tearing down %s", self._testMethodName)
-        boom.set_boot_path(BOOT_ROOT_TEST)
-        self._clear_fstab()
```

Comment:
We definitely want these to run during `tearDown()` to re-set the state between tests. Ideally we'd have `setUp()` and `tearDown()` defined only in the base class.